recipe/meta.yaml

```diff
@@ -1,21 +1,26 @@
+{% set version = "1.900.1" %}
+{% set jpeg_version = os.environ.get('JPEG_VERSION', '9b')|string %}
+{% set build_number = "4" %}
+
 package:
     name: jasper
-    version: 1.900.1
+    version: {{ version }}
 
 source:
-    fn: jasper-1.900.1.zip
-    url: http://www.ece.uvic.ca/~frodo/jasper/software/jasper-1.900.1.zip
+    fn: jasper-{{ version }}.zip
+    url: http://www.ece.uvic.ca/~frodo/jasper/software/jasper-{{ version }}.zip
     sha1: 9c5735f773922e580bf98c7c7dfda9bbed4c5191
 
 build:
-    number: 3
+    number: {{ build_number }}
+    string: jpeg{{ jpeg_version }}_{{ build_number }}
     skip: True  # [win]
 
 requirements:
     build:
-        - jpeg 9*
+        - jpeg {{ jpeg_version }}
```

Comment:
I do re-build `jasper` for our local builds with `jpeg 8*` using a script similar to pin_slow.py used in conda-forge. It is convenient for me, because I control all my pinnins with one file, and non-intrusive for conda-forge. Maybe you should consider using that too.